services/QuillLMS/client/app/bundles/Shared/components/shared/tooltip.tsx

```diff
@@ -25,22 +29,29 @@ export class Tooltip extends React.Component<TooltipProps, {}> {
     this.hideTooltip = this.hideTooltip.bind(this)
   }
 
+
   componentWillUnmount() {
     document.removeEventListener('click', this.hideTooltipOnClick, false)
   }
 
   hideTooltipOnClick(e) {
+    const { setNavigationUrl, toggleWarningModal } = this.props;
     if (this.tooltip && e.target !== this.tooltip && e.target !== this.tooltipTrigger) {
       this.hideTooltip()
     }
+    if(e.target.href) {
+      e.preventDefault();
+      setNavigationUrl(e.target.href)
+      toggleWarningModal();
+    }
```

Comment:
This logic is too specific for this shared component to know about - we can refactor this so you're just passing an `onClick` handler from the parent which then does all this logic for you, instead of passing in these props that are hyperspecific to this one implementation.